Detectors/GlobalTracking/include/GlobalTracking/TrackCuts.h

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file TrackCuts.h
+/// \brief Class to perform track cuts
+/// \author amelia.lindner@cern.ch
+
+#ifndef ALICEO2_TRACKCUTS_H
+#define ALICEO2_TRACKCUTS_H
+
+#include "Framework/Logger.h"
+#include "Framework/DataTypes.h"
+#include "ReconstructionDataFormats/Track.h"
+#include "DataFormatsGlobalTracking/RecoContainer.h"
+#include "ReconstructionDataFormats/MatchInfoTOF.h"
+#include "DataFormatsTPC/TrackTPC.h"
+#include "DataFormatsTRD/TrackTRD.h"
+#include "GlobalTracking/TrackMethods.h"
+#include "DetectorsBase/Propagator.h"
+#include "ReconstructionDataFormats/VtxTrackIndex.h"
+#include <set>
+#include <vector>
+#include "Rtypes.h"
+#include <gsl/span>
+namespace o2
+{
+using GID = o2::dataformats::GlobalTrackID;
+using GIndex = o2::dataformats::VtxTrackIndex;
+class TrackCuts
+{
+ public:
+  //////////////////////////////// O2 ////////////////////////////////
+  bool isSelected(GID trackIndex, o2::globaltracking::RecoContainer& data)
+  {
+    o2::track::TrackParCov trk;
+    auto contributorsGID = data.getSingleDetectorRefs(trackIndex);
+    auto src = trackIndex.getSource();                       // make selections depending on track source
+    if (contributorsGID[GIndex::Source::ITS].isIndexSet()) { // ITS tracks selection
+      const auto& itsTrk = data.getITSTrack(contributorsGID[GID::ITS]);
+      int ITSnClusters = itsTrk.getNClusters();
+      float ITSchi2 = itsTrk.getChi2();
+      float itsChi2NCl = ITSnClusters != 0 ? ITSchi2 / (float)ITSnClusters : 0;
+      uint8_t itsClusterMap = itsTrk.getPattern();
+      if (itsTrk.getPt() >= mMinPt && itsTrk.getPt() <= mMaxPt &&
+          itsTrk.getEta() >= mMinEta && itsTrk.getEta() <= mMaxEta &&
+          ITSnClusters >= mMinNClustersITS &&
+          itsChi2NCl <= mMaxChi2PerClusterITS &&
+          TrackMethods::FulfillsITSHitRequirements(itsClusterMap, mRequiredITSHits)) {
+        return true;
```

Comment:
The track may have contributions from different detectors, particularly, here you are checking the ITS part. but why do return success once the ITS selection is passed? It still may fail on e.g. TPC contribution checks. Only in case of failure on ITS you may stop checking.

Reply:
I changed this, now it just returns false if the checks are not passed